engine/src/main/java/org/terasology/world/block/family/CeilingSupportingHorizontalFamily.java

```diff
@@ -122,13 +122,16 @@ private Block transformBlock(
 
     @Override
     public Block getBlockForPlacement(BlockPlacementData data) {
-        Side blockDirection = Side.inDirection(-data.viewingDirection.x(), 0, -data.viewingDirection.z());
+        Side mainSide = Side.TOP;
 
-        if (data.attachmentSide == Side.BOTTOM) {
-            return blocks.get(ExtendedSide.getExtendedSideFor(Side.BOTTOM, blockDirection));
-        } else {
-            return blocks.get(ExtendedSide.getExtendedSideFor(Side.TOP, blockDirection));
+        boolean upsideDownPlacement = data.attachmentSide == Side.BOTTOM
+                || data.attachmentSide != Side.TOP && data.relativeAttachmentPosition.y() > 0.5;
+        if (upsideDownPlacement) {
+            mainSide = Side.BOTTOM;
         }
```

Comment:
Instead of declaring `mainSide` and then mutating the value you could do the check first and just do a single-time assignment:

```suggestion
        final Side mainSide = upsideDownPlacement ? Side.BOTTOM : Side.TOP;
```

Reply:
Yes, much better.